### PR TITLE
system var: Active PD follower for handling region request

### PR DESCRIPTION
**Pull request by CabinfeverB**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tune-region-performance.md [hunk 1]
```diff
@@ -33,4 +33,14 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
+
+## 通过 Active PD follower 提升 PD Region 路由信息查询的服务能力
+
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。
+
+当集群 Region 数量较多时，PD leader 本身会由于处理心跳和调度有较大的开销，造成 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多， Region 信息请求的并发量较大，PD leader CPU 压力变得更大，甚至可能造成 PD 服务不可用。
+
+为了保证高可用，PD leader 会将 Region 信息实时同步给 follower，follower 会在内存中维护保存 Region 信息，这就使得 PD follower 同样有处理 Region 信息请求的能力。可以通过设置 TiDB 参数 [`pd_enable_follower_handle_region`](/system-variables.md#pd_enable_follower_handle_region-从-v760-版本开始引入) 开启 Active PD follower 特性，开启该特性之后，TiDB 在获取 Region 信息 时会将请求均匀地发送到所有 PD 节点上，PD follower 也可以直接处理 Region 请求，从而降低 PD leader 的 CPU 压力。
```
[the remark this answers is not in the document]
Comment (CabinfeverB) [reply tune-region-performance.md:46]: 好的

File: system-variables.md [hunk 2]
```diff
@@ -2293,6 +2293,18 @@ Query OK, 0 rows affected (0.09 sec)
 >
 > 如果 PD leader 的 TSO RPC 延迟升高，但其现象并非由 CPU 使用率达到瓶颈而导致（可能存在网络等问题），此时，打开 TSO Follower Proxy 可能会导致 TiDB 的语句执行延迟上升，从而影响集群的 QPS 表现。
 
+### `pd_enable_follower_handle_region` <span class="version-mark">从 v7.6.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline system-variables.md:2296]: Please keep the `system-variables.md` document in ascending order.

File: system-variables.md [hunk 3]
```diff
@@ -2293,6 +2293,18 @@ Query OK, 0 rows affected (0.09 sec)
...
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用来开启 Active PD Follower 特性（目前只适用于 Region 信息的相关请求）。当该值为 `OFF` 时，TiDB 仅会从 PD leader 获取 Region 信息。开启该特性之后，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，PD follower 也可以直接处理 Region 请求，从而降低 PD leader 的 CPU 压力。
```
Comment (Oreoxmt) [inline system-variables.md:2303]:
```suggestion
- 这个变量用于控制是否开启 Active PD Follower 特性（目前只适用于 Region 信息的相关请求）。当该值为 `OFF` 时，TiDB 仅从 PD leader 获取 Region 信息。当该值为 `ON` 时，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，PD follower 也可以直接处理 Region 请求，从而减轻 PD leader 的 CPU 压力。
```

File: system-variables.md [hunk 4]
```diff
@@ -2293,6 +2293,18 @@ Query OK, 0 rows affected (0.09 sec)
...
+- 适合开启 Active PD Follower 的场景：
+    * 集群 Region 数量较多，PD leader 本身由于处理心跳和调度的开销大，CPU 资源紧张。
+    * 集群中的 TiDB 实例数量较多，Region 信息请求的并发量较大，PD leader CPU 压力大。
```
Comment (Oreoxmt) [inline system-variables.md:2306]: “数量较多”比较模糊，是否可以给出具体的数值范围？

Comment (Oreoxmt) [inline system-variables.md:2306]:
```suggestion
    * 集群 Region 数量较多，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。
    * 集群中 TiDB 实例数量较多，Region 信息请求并发量较大，导致 PD leader CPU 压力大。
```

File: system-variables.md [hunk 5]
```diff
@@ -2293,6 +2293,18 @@ Query OK, 0 rows affected (0.09 sec)
 >
 > 如果 PD leader 的 TSO RPC 延迟升高，但其现象并非由 CPU 使用率达到瓶颈而导致（可能存在网络等问题），此时，打开 TSO Follower Proxy 可能会导致 TiDB 的语句执行延迟上升，从而影响集群的 QPS 表现。
 
+### `pd_enable_follower_handle_region` <span class="version-mark">从 v7.6.0 版本开始引入</span>
+
```
Comment (Oreoxmt) [inline system-variables.md:2297]:
```suggestion

> **警告：**
>
> Active PD Follower 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。

```

File: tune-region-performance.md [hunk 6]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
 
-Region 调大以后，如需进一步提高查询的并发度，可以设置 [`coprocessor.enable-region-bucket`](/tikv-configuration-file.md#enable-region-bucket-从-v610-版本开始引入) 为 `true`。这个配置会将每个 Region 划分为更小的区间 bucket，并且以这个更小的区间作为并发查询单位，以提高扫描数据的并发度。bucket 的大小通过 [`coprocessor.region-bucket-size`](/tikv-configuration-file.md#region-bucket-size-从-v610-版本开始引入) 来控制。
\ No newline at end of file
+Region 调大以后，如需进一步提高查询的并发度，可以设置 [`coprocessor.enable-region-bucket`](/tikv-configuration-file.md#enable-region-bucket-从-v610-版本开始引入) 为 `true`。这个配置会将每个 Region 划分为更小的区间 bucket，并且以这个更小的区间作为并发查询单位，以提高扫描数据的并发度。bucket 的大小通过 [`coprocessor.region-bucket-size`](/tikv-configuration-file.md#region-bucket-size-从-v610-版本开始引入) 来控制。
+
+## 通过 Active PD follower 提升 PD Region 路由信息查询的服务能力
+
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。
```
Comment (Oreoxmt) [inline tune-region-performance.md:42]:
```suggestion
> Active PD Follower 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: tune-region-performance.md [hunk 7]
```diff
@@ -5,7 +5,7 @@ summary: 了解如何通过调整 Region 大小等方法对 Region 进行性能
 
 # Region 性能调优
 
-本文介绍了如何通过调整 Region 大小等方法对 Region 进行性能调优以及如何在大 Region 下使用 bucket 进行并发查询优化。
+本文介绍了如何通过调整 Region 大小等方法对 Region 进行性能调优以及如何在大 Region 下使用 bucket 进行并发查询优化。此外，本文同时介绍了通过 Active PD Follower 特性来提升 PD 对 TiDB 节点提供 Region 信息的服务能力。
```
Comment (Oreoxmt) [inline tune-region-performance.md:8]:
```suggestion
本文介绍了如何通过调整 Region 大小等方法对 Region 进行性能调优以及如何在大 Region 下使用 bucket 进行并发查询优化。同时，本文还介绍了通过开启 Active PD Follower 特性提升 PD 对 TiDB 节点提供 Region 信息的服务能力。
```

File: tune-region-performance.md [hunk 8]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
 >
 > 当前该功能为实验特性，不建议在生产环境中使用。
 
-Region 调大以后，如需进一步提高查询的并发度，可以设置 [`coprocessor.enable-region-bucket`](/tikv-configuration-file.md#enable-region-bucket-从-v610-版本开始引入) 为 `true`。这个配置会将每个 Region 划分为更小的区间 bucket，并且以这个更小的区间作为并发查询单位，以提高扫描数据的并发度。bucket 的大小通过 [`coprocessor.region-bucket-size`](/tikv-configuration-file.md#region-bucket-size-从-v610-版本开始引入) 来控制。
\ No newline at end of file
+Region 调大以后，如需进一步提高查询的并发度，可以设置 [`coprocessor.enable-region-bucket`](/tikv-configuration-file.md#enable-region-bucket-从-v610-版本开始引入) 为 `true`。这个配置会将每个 Region 划分为更小的区间 bucket，并且以这个更小的区间作为并发查询单位，以提高扫描数据的并发度。bucket 的大小通过 [`coprocessor.region-bucket-size`](/tikv-configuration-file.md#region-bucket-size-从-v610-版本开始引入) 来控制。
+
+## 通过 Active PD follower 提升 PD Region 路由信息查询的服务能力
```
Comment (Oreoxmt) [inline tune-region-performance.md:38]:
```suggestion
## 通过 Active PD Follower 提升 PD Region 路由信息查询的服务能力
```

File: tune-region-performance.md [hunk 9]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
+
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。
+
+当集群 Region 数量较多时，PD leader 本身会由于处理心跳和调度有较大的开销，造成 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多， Region 信息请求的并发量较大，PD leader CPU 压力变得更大，甚至可能造成 PD 服务不可用。
```
Comment (Oreoxmt) [inline tune-region-performance.md:44]:
```suggestion
当集群的 Region 数量较多时，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多，Region 信息请求并发量较大，PD leader CPU 压力变得更大，可能会造成 PD 服务不可用。
```

File: tune-region-performance.md [hunk 10]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
+
+为了保证高可用，PD leader 会将 Region 信息实时同步给 follower，follower 会在内存中维护保存 Region 信息，这就使得 PD follower 同样有处理 Region 信息请求的能力。可以通过设置 TiDB 参数 [`pd_enable_follower_handle_region`](/system-variables.md#pd_enable_follower_handle_region-从-v760-版本开始引入) 开启 Active PD follower 特性，开启该特性之后，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，PD follower 也可以直接处理 Region 请求，从而降低 PD leader 的 CPU 压力。
```
Comment (Oreoxmt) [inline tune-region-performance.md:46]:
```suggestion
为确保服务的高可用性，PD leader 会将 Region 信息实时同步给 follower。PD follower 在内存中维护并保存这些 Region 信息，从而具备处理 Region 信息请求的能力。你可以通过设置系统变量 [`pd_enable_follower_handle_region = ON`](/system-variables.md#pd_enable_follower_handle_region-从-v760-版本开始引入) 开启 Active PD Follower 特性。启用该特性后，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，使 PD follower 也可以直接处理 Region 请求，从而减轻 PD leader 的 CPU 压力。
```

File: tune-region-performance.md [hunk 11]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
+
+得益于 PD 自身维护了 Region 同步流的状态以及 TiKV client-go 的 fallback 机制，能够保证 TiDB 中的 Region 信息都是最新的。当 PD leader 和 follower 之间网络不稳定、follower 不可用等情况导致 region 同步流断开时，PD follower 会拒绝 Region 请求，此时 TiDB 会自发地向 PD leader 重试，并将 follower 暂时置为不可用状态。但在网络稳定时，leader 和 follower 之间的同步也仍然存在时间差，此时从 follower 获取到的个别 Region 信息仍可能是过期的。当这些 Region 对应的 KV Request 失败后，TiDB 会向 PD leader 再次获取最新的 Region 信息，然后再次发送 KV Request 到 TiKV。
```
Comment (Oreoxmt) [inline tune-region-performance.md:48]:
```suggestion
PD 通过维护 Region 同步流的状态，并结合 TiKV client-go 的 fallback 机制，确保 TiDB 中的 Region 信息始终是最新的。若 PD leader 与 follower 之间网络不稳定或 follower 不可用， 导致 Region 同步流断开，PD follower 将拒绝处理 Region 请求。此时，TiDB 会自动向 PD leader 重试请求，并将该 follower 暂时标记为不可用状态。但是，在网络稳定时，由于 leader 和 follower 之间的同步可能存在延迟，从 follower 获取的部分 Region 信息可能是过时的。在这种情况下，如果 Region 对应的 KV Request 失败，TiDB 会重新向 PD leader 请求最新的 Region 信息，并再次向 TiKV 发送 KV Request。
```

**Review by rleungx — APPROVED** [review]

File: system-variables.md (hunk 4 above)
Comment (CabinfeverB) [reply system-variables.md:2306]: 这里不太好给出具体范围，和集群整体大小，PD 规格都有关系

File: system-variables.md [hunk 12]
```diff
@@ -2284,15 +2300,15 @@ Query OK, 0 rows affected (0.09 sec)
...
+- 这个变量用于控制是否开启 TSO Follower Proxy 特性。当该值为 `OFF` 时，TiDB 仅会从 PD leader 获取 TSO。当该值为 `ON` 时，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而减轻 PD leader 的 CPU 压力。
 - 适合开启 TSO Follower Proxy 的场景：
     * PD leader 因高压力的 TSO 请求而达到 CPU 瓶颈，导致 TSO RPC 请求的延迟较高。
     * 集群中的 TiDB 实例数量较多，且调高 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v530-版本开始引入) 并不能缓解 TSO RPC 请求延迟高的问题。
 
 > **注意：**
 >
 > 如果 PD leader 的 TSO RPC 延迟升高，但其现象并非由 CPU 使用率达到瓶颈而导致（可能存在网络等问题），此时，打开 TSO Follower Proxy 可能会导致 TiDB 的语句执行延迟上升，从而影响集群的 QPS 表现。
-
+
```
Comment (Oreoxmt) [inline system-variables.md:2311]:
```suggestion

```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 13]
```diff
@@ -2284,15 +2300,14 @@ Query OK, 0 rows affected (0.09 sec)
...
 - 默认值：`OFF`
-- 这个变量用来开启 TSO Follower Proxy 特性。当该值为 `OFF` 时，TiDB 仅会从 PD leader 获取 TSO。开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。
+- 这个变量用于控制是否开启 TSO Follower Proxy 特性。当该值为 `OFF` 时，TiDB 仅会从 PD leader 获取 TSO。当该值为 `ON` 时，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而减轻 PD leader 的 CPU 压力。
 - 适合开启 TSO Follower Proxy 的场景：
     * PD leader 因高压力的 TSO 请求而达到 CPU 瓶颈，导致 TSO RPC 请求的延迟较高。
     * 集群中的 TiDB 实例数量较多，且调高 [`tidb_tso_client_batch_max_wait_time`](/system-variables.md#tidb_tso_client_batch_max_wait_time-从-v530-版本开始引入) 并不能缓解 TSO RPC 请求延迟高的问题。
 
 > **注意：**
 >
 > 如果 PD leader 的 TSO RPC 延迟升高，但其现象并非由 CPU 使用率达到瓶颈而导致（可能存在网络等问题），此时，打开 TSO Follower Proxy 可能会导致 TiDB 的语句执行延迟上升，从而影响集群的 QPS 表现。
```
Comment (Oreoxmt) [inline system-variables.md:2310]:
```suggestion
> 如果 PD leader 的 TSO RPC 延迟升高，但其现象并非由 CPU 使用率达到瓶颈而导致（可能存在网络等问题），此时，打开 TSO Follower Proxy 可能会导致 TiDB 的语句执行延迟上升，从而影响集群的 QPS 表现。

```

File: tune-region-performance.md [hunk 14]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
\ No newline at end of file
+Region 调大以后，如需进一步提高查询的并发度，可以设置 [`coprocessor.enable-region-bucket`](/tikv-configuration-file.md#enable-region-bucket-从-v610-版本开始引入) 为 `true`。这个配置会将每个 Region 划分为更小的区间 bucket，并且以这个更小的区间作为并发查询单位，以提高扫描数据的并发度。bucket 的大小通过 [`coprocessor.region-bucket-size`](/tikv-configuration-file.md#region-bucket-size-从-v610-版本开始引入) 来控制。
+
+## 通过 Active PD Follower 提升 PD Region 路由信息查询的服务能力
+
+> **警告：**
+>
+> Active PD Follower 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+当集群的 Region 数量较多时，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多， Region 信息请求并发量较大，PD leader CPU 压力变得更大，可能会造成 PD 服务不可用。
```
Comment (Oreoxmt) [inline tune-region-performance.md:44]:
```suggestion
当集群的 Region 数量较多时，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多，Region 信息请求并发量较大，PD leader CPU 压力变得更大，可能会造成 PD 服务不可用。
```

File: tune-region-performance.md [hunk 15]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
+
+> **警告：**
+>
+> Active PD Follower 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+当集群的 Region 数量较多时，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多，Region 信息请求并发量较大，PD leader CPU 压力变得更大，可能会造成 PD 服务不可用。
+
+为确保服务的高可用性，PD leader 会将 Region 信息实时同步给 follower。PD follower 在内存中维护保存 Region 信息，从而具备处理 Region 信息请求的能力。你可以通过设置系统变量 [`pd_enable_follower_handle_region`](/system-variables.md#pd_enable_follower_handle_region-从-v760-版本开始引入) 开启 Active PD Follower 特性。启用该特性后，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，使 PD follower 也可以直接处理 Region 请求，从而降低减轻 PD leader 的 CPU 压力。
+
+PD 通过维护 Region 同步流的状态，并结合 TiKV client-go 的 fallback 机制，确保 TiDB 中的 Region 信息始终是最新的。若 PD leader 与 follower 之间网络不稳定或 follower 不可用， 导致 Region 同步流断开，PD follower 将拒绝处理 Region 请求。此时，TiDB 会自动向 PD leader 重试请求，并将该 follower 暂时标记为不可用状态。但是，在网络稳定时，由于 leader 和 follower 之间的同步可能存在延迟，从 follower 获取的部分 Region 信息可能是过时的。在这种情况下，如果 Region 对应的 KV Request 失败，TiDB 会重新向 PD leader 请求最新的 Region 信息，并再次向 TiKV 发送 KV Request。
```
Comment (Oreoxmt) [inline tune-region-performance.md:48]:
```suggestion
PD 通过维护 Region 同步流的状态，并结合 TiKV client-go 的 fallback 机制，确保 TiDB 中的 Region 信息始终是最新的。若 PD leader 与 follower 之间网络不稳定或 follower 不可用，导致 Region 同步流断开，PD follower 将拒绝处理 Region 请求。此时，TiDB 会自动向 PD leader 重试请求，并将该 follower 暂时标记为不可用状态。但是，在网络稳定时，由于 leader 和 follower 之间的同步可能存在延迟，从 follower 获取的部分 Region 信息可能是过时的。在这种情况下，如果 Region 对应的 KV Request 失败，TiDB 会重新向 PD leader 请求最新的 Region 信息，并再次向 TiKV 发送 KV Request。
```

File: system-variables.md [hunk 16]
```diff
@@ -711,6 +711,22 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
...
+> **警告：**
+>
+> Active PD Follower 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用于控制是否开启 Active PD Follower 特性（目前只适用于 Region 信息的相关请求）。当该值为 `OFF` 时，TiDB 仅从 PD leader 获取 Region 信息。当该值为 `ON` 时，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，PD follower 也可以直接处理 Region 请求，从而减轻 PD leader 的 CPU 压力。
```
Comment (qiancai) [inline system-variables.md:725]:
```suggestion
- 这个变量用于控制是否开启 Active PD Follower 特性（目前该特性只适用于处理获取 Region 信息的相关请求）。当该值为 `OFF` 时，TiDB 仅从 PD leader 获取 Region 信息。当该值为 `ON` 时，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，因此 PD follower 也可以处理 Region 信息请求，从而减轻 PD leader 的 CPU 压力。
```

File: system-variables.md [hunk 17]
```diff
@@ -711,6 +711,22 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
...
+- 适合开启 Active PD Follower 的场景：
+    * 集群 Region 数量较多，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。
+    * 集群中 TiDB 实例数量较多，Region 信息请求并发量较大，PD leader CPU 压力大。
```
Comment (qiancai) [inline system-variables.md:728]:
```suggestion
    - 集群 Region 数量较多，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。
    - 集群中 TiDB 实例数量较多，Region 信息请求并发量较大，PD leader CPU 压力大。
```

File: system-variables.md [hunk 18]
```diff
@@ -711,6 +711,22 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 范围：`[0, 4294967295]`
 - 该变量用于建立密码重用策略，使 TiDB 基于经过时间限制密码重复使用。该变量默认值为 0，表示禁用基于密码经过时间的密码重用策略。当设置该变量为一个正整数 N 时，表示不允许重复使用最近 N 天内使用过的密码。
 
+### `pd_enable_follower_handle_region` <span class="version-mark">从 v7.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> Active PD Follower 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```
Comment (qiancai) [inline system-variables.md:718]:
```suggestion
> [Active PD Follower](/tune-region-performance.md#通过-active-pd-follower-提升-pd-region-路由信息查询的服务能力) 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

File: system-variables.md [hunk 19]
```diff
@@ -2284,7 +2300,7 @@ Query OK, 0 rows affected (0.09 sec)
 - 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
 - 类型：布尔型
 - 默认值：`OFF`
-- 这个变量用来开启 TSO Follower Proxy 特性。当该值为 `OFF` 时，TiDB 仅会从 PD leader 获取 TSO。开启该特性之后，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而降低 PD leader 的 CPU 压力。
+- 这个变量用于控制是否开启 TSO Follower Proxy 特性。当该值为 `OFF` 时，TiDB 仅会从 PD leader 获取 TSO。当该值为 `ON` 时，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，通过 PD follower 转发 TSO 请求，从而减轻 PD leader 的 CPU 压力。
```
Comment (qiancai) [inline system-variables.md:2303]:
```suggestion
- 这个变量用于控制是否开启 TSO Follower Proxy 特性。当该值为 `OFF` 时，TiDB 仅会从 PD leader 获取 TSO。当该值为 `ON` 时，TiDB 在获取 TSO 时会将请求均匀地发送到所有 PD 节点上，因此 PD follower 也可以处理 TSO 请求，从而减轻 PD leader 的 CPU 压力。
```

File: tune-region-performance.md [hunk 20]
```diff
@@ -5,7 +5,7 @@ summary: 了解如何通过调整 Region 大小等方法对 Region 进行性能
 
 # Region 性能调优
 
-本文介绍了如何通过调整 Region 大小等方法对 Region 进行性能调优以及如何在大 Region 下使用 bucket 进行并发查询优化。
+本文介绍了如何通过调整 Region 大小等方法对 Region 进行性能调优以及如何在大 Region 下使用 bucket 进行并发查询优化。同时，本文还介绍了通过开启 Active PD Follower 特性提升 PD 对 TiDB 节点提供 Region 信息的服务能力。
```
Comment (qiancai) [inline tune-region-performance.md:8]:
```suggestion
本文介绍了如何通过调整 Region 大小等方法对 Region 进行性能调优以及如何在大 Region 下使用 bucket 进行并发查询优化。同时，本文还介绍了通过开启 Active PD Follower 特性来提升 PD 为 TiDB 节点提供 Region 信息的服务能力。
```

File: tune-region-performance.md [hunk 21]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
\ No newline at end of file
+Region 调大以后，如需进一步提高查询的并发度，可以设置 [`coprocessor.enable-region-bucket`](/tikv-configuration-file.md#enable-region-bucket-从-v610-版本开始引入) 为 `true`。这个配置会将每个 Region 划分为更小的区间 bucket，并且以这个更小的区间作为并发查询单位，以提高扫描数据的并发度。bucket 的大小通过 [`coprocessor.region-bucket-size`](/tikv-configuration-file.md#region-bucket-size-从-v610-版本开始引入) 来控制。
+
+## 通过 Active PD Follower 提升 PD Region 路由信息查询的服务能力
+
+> **警告：**
+>
+> Active PD Follower 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+当集群的 Region 数量较多时，PD leader 由于处理心跳和调度任务的开销大，导致 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多，Region 信息请求并发量较大，PD leader CPU 压力变得更大，可能会造成 PD 服务不可用。
```
Comment (qiancai) [inline tune-region-performance.md:44]:
```suggestion
当集群的 Region 数量较多时，PD leader 处理心跳和调度任务的开销也较大，可能导致 CPU 资源紧张。如果同时集群中的 TiDB 实例数量较多，Region 信息请求并发量较大，PD leader CPU 压力将变得更大，可能会造成 PD 服务不可用。
```

File: tune-region-performance.md [hunk 22]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
+
+为确保服务的高可用性，PD leader 会将 Region 信息实时同步给 follower。PD follower 在内存中维护保存 Region 信息，从而具备处理 Region 信息请求的能力。你可以通过设置系统变量 [`pd_enable_follower_handle_region`](/system-variables.md#pd_enable_follower_handle_region-从-v760-版本开始引入) 开启 Active PD Follower 特性。启用该特性后，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，使 PD follower 也可以直接处理 Region 请求，从而降低减轻 PD leader 的 CPU 压力。
```
Comment (qiancai) [inline tune-region-performance.md:46]:
```suggestion
为确保服务的高可用性，PD leader 会将 Region 信息实时同步给 PD follower。PD follower 在内存中维护保存 Region 信息，从而具备处理 Region 信息请求的能力。你可以通过设置系统变量 [`pd_enable_follower_handle_region`](/system-variables.md#pd_enable_follower_handle_region-从-v760-版本开始引入) 开启 Active PD Follower 特性。启用该特性后，TiDB 在获取 Region 信息时会将请求均匀地发送到所有 PD 节点上，使 PD follower 也可以直接处理 Region 请求，从而降低减轻 PD leader 的 CPU 压力。
```

File: tune-region-performance.md [hunk 23]
```diff
@@ -33,4 +33,16 @@ Region 的大小可以通过 [`coprocessor.region-split-size`](/tikv-configurati
...
+
+PD 通过维护 Region 同步流的状态，并结合 TiKV client-go 的 fallback 机制，确保 TiDB 中的 Region 信息始终是最新的。若 PD leader 与 follower 之间网络不稳定或 follower 不可用，导致 Region 同步流断开，PD follower 将拒绝处理 Region 请求。此时，TiDB 会自动向 PD leader 重试请求，并将该 follower 暂时标记为不可用状态。但是，在网络稳定时，由于 leader 和 follower 之间的同步可能存在延迟，从 follower 获取的部分 Region 信息可能是过时的。在这种情况下，如果 Region 对应的 KV Request 失败，TiDB 会重新向 PD leader 请求最新的 Region 信息，并再次向 TiKV 发送 KV Request。
```
Comment (qiancai) [inline tune-region-performance.md:48]:
```suggestion
PD 通过维护 Region 同步流的状态，并结合 TiKV client-go 的 fallback 机制，确保 TiDB 中的 Region 信息始终是最新的。

- 当 PD leader 与 follower 之间网络不稳定或 follower 不可用导致 Region 同步流断开时，PD follower 将拒绝处理 Region 请求。此时，TiDB 会自动向 PD leader 重试请求，并将该 follower 暂时标记为不可用状态。
- 当网络稳定时，由于 PD leader 和 follower 之间的同步可能存在延迟，从 follower 获取的部分 Region 信息可能是过时的。在这种情况下，如果 Region 对应的 KV Request 失败，TiDB 会重新向 PD leader 请求最新的 Region 信息，并再次向 TiKV 发送 KV Request。
```

File: system-variables.md [hunk 24]
```diff
@@ -711,6 +711,22 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - 范围：`[0, 4294967295]`
 - 该变量用于建立密码重用策略，使 TiDB 基于经过时间限制密码重复使用。该变量默认值为 0，表示禁用基于密码经过时间的密码重用策略。当设置该变量为一个正整数 N 时，表示不允许重复使用最近 N 天内使用过的密码。
 
+### `pd_enable_follower_handle_region` <span class="version-mark">从 v7.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> [Active PD Follower](/tune-region-performance.md#通过-active-pd-follower-提升-pd-region-路由信息查询的服务能力) 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```
Comment (Oreoxmt) [inline system-variables.md:718]:
```suggestion
> [Active PD Follower](/tune-region-performance.md#通过-active-pd-follower-提升-pd-region-信息查询服务的扩展能力) 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**